source/qb64pe.bas

```diff
@@ -3637,23 +3637,34 @@ DO
     'also check if such a handle was defined and mark it as used then.
     IF INSTR(a3u$, qb64prefix$ + "EMBEDDED$") > 0 THEN
         spo = INSTR(a3u$, qb64prefix$ + "EMBEDDED$"): ket = 0
+        ppo = INSTR(spo + 1, a3u$, "("): IF ppo = 0 THEN ppo = LEN(a3u$) + 1
+        cle = LEN(qb64prefix$ + "EMBEDDED$")
+        gap$ = _TRIM$(MID$(a3u$, spo + cle, ppo - spo - cle))
+        IF gap$ <> "" GOTO skipEflCheck 'not a function call
         DO
-            'ignore everything after comment (' or REM)
+            'ignore everything in strings or after comment (' or REM)
             rsq = 0 'quote flag
             FOR i = ket + 1 TO spo
                 IF ASC(a3u$, i) = 34 THEN rsq = NOT rsq
                 IF ASC(a3u$, i) = 39 AND NOT rsq THEN EXIT DO
-                IF MID$(a3u$, i, 3) = "REM" AND NOT rsq THEN EXIT DO
+                IF MID$(a3u$, i, 3) = "REM" AND NOT rsq THEN
+                    IF i > 1 THEN rsf = ASC(a3u$, i - 1): ELSE rsf = 32 'front char
+                    rsr = ASC(a3u$, i + 3) 'rear char
+                    IF (rsf = 9 OR rsf = 32 OR rsf = 58) AND (rsr = 9 OR rsr = 32) THEN EXIT DO 'is real REM
+                END IF
             NEXT i
+            IF rsq THEN EXIT DO 'inside a quoted string
```

Comment:
Along with the `skipElfCheck` jump before the loop, exiting the loop seems to cause the rest of the line to get ignored, even if there's more instances to check. Ex. In lines like `Print "_Embedded$"; _Embedded$("foo")` and `Print "_Embedded$("; _Embedded$("foo")` the second `_Embedded$` call is not getting checked even though it's real.

Reply:
Sure, as above, the check must go inside the loop.